homeassistant/components/sky_hub/device_tracker.py

```diff
@@ -44,11 +44,22 @@ def __init__(self, hub):
     async def async_scan_devices(self):
         """Scan for new devices and return a list with found device IDs."""
         await self._async_update_info()
-        return list(self.last_results)
+        return [device.mac for device in self.last_results]
 
     async def async_get_device_name(self, device):
         """Return the name of the given device."""
-        return self.last_results.get(device)
+        name = next(
+            (result.name for result in self.last_results if result.mac == device),
+            None,
+        )
+        return name
+
+    async def async_get_extra_attributes(self, device):
+        """Get extra attributes of a device."""
+        device = next(
+            (result for result in self.last_results if result.mac == device), None
+        )
+        return device.asdict()
```

Comment:
```suggestion
        if device is None:
            return device

        return device.asdict()
```

Reply:
Please explicitly write out what data we're returning here. We want to see the keys and check the formatting.